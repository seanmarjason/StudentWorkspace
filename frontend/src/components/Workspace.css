.workspace {
  display: flex;
  flex-wrap: wrap;
  align-content: flex-start;
  height: 90vh;
}

.workspace h1 {
  width: 100vw;
  text-align: center;
}

.workspace h2 {
  width: 100vw;
  text-align: center;
  padding-bottom: 5vh;
}

.footer {
  height: 10vh;
  width: 100vw;
}