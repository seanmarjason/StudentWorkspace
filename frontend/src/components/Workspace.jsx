import { useState, useEffect } from 'react';
import axios from 'axios';
import './Workspace.css';
import { Section } from './Section';
import { AddSection } from './AddSection';

const Workspace = ({ groupId }) => {

  const [workspaceData, setWorkspaceData] = useState(null);

  useEffect(() => {
    const url = `http://localhost:3000/workspaces/${groupId.toString()}`

    axios.get(url)
      .then(response => setWorkspaceData(response.data))
      .catch(error => console.log(`Error: ${error}`))
  }, [groupId]);

  const moveSection = (sectionName, direction) => {
    const url = `http://localhost:3000/workspaces/${groupId.toString()}/${sectionName}/${direction}`
    
    axios.patch(url)
      .then(response => setWorkspaceData(response.data))
      .catch(error => console.log(`Error: ${error}`))
  }

  return (
    <div className="workspace">
      {
        workspaceData ?
          <>
            <h1>{workspaceData.name}</h1>
            <h2>{workspaceData.groupReference}</h2>
            <div className="workspaceSections">
              {
                workspaceData.sections.map((section) =>
                  <div className="sectionContainer" id={section.name} key={section.name}>
                    <Section
                      groupId={groupId}
                      sectionName={section.name}
                    />
                    <div className="moveButtons">
                      <span onClick={() => moveSection(section.name, 'up')}>&#8593;</span>
                      <br />
                      <br />
                      <span onClick={() => moveSection(section.name, 'down')}>&#8595;</span>
                    </div>
                  </div>
                )
              }
            </div>
          <AddSection 
            groupId={groupId}
            setWorkspaceData={setWorkspaceData}
          />
          </>
          :
          <p>Loading...</p>
        }
        <div className="footer"></div>
        </div>
  )
}


export { Workspace }
