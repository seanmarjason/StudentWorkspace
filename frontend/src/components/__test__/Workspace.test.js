import {render, screen, waitFor } from '@testing-library/react';
import axios from 'axios';
import { Workspace } from '../Workspace';

const workspace = {
  data: {
    "group_id": "100",
    "name": "Test Workspace 1",
    "groupReference": "Group ABC",
    "groupMembers": [],
    "sections": [
      "research",
      "deliverables"
    ]
  }
}

jest.mock('axios');

jest.mock('../Section', () => ({
  Section: () => <div className="section" />
  }
));

describe('Workspace' , () => {
  it('should match snapshot', async () => {
    axios.get.mockResolvedValueOnce(workspace);

    const view = render(<Workspace groupId={'100'} />)

    await waitFor(() => {
      screen.getByText('Test Workspace 1')
    });

    expect(view.asFragment()).toMatchSnapshot();
  });
  
})
  