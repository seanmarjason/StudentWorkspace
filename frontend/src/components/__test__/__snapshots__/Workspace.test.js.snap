// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="workspace"
  >
    <h1>
      Test Workspace 1
    </h1>
    <h2>
      Group ABC
    </h2>
    <div
      class="workspaceSections"
    >
      <div
        class="sectionContainer"
      >
        <div
          class="section"
        />
        <div
          class="moveButtons"
        >
          <span>
            ↑
          </span>
          <br />
          <br />
          <span>
            ↓
          </span>
        </div>
      </div>
      <div
        class="sectionContainer"
      >
        <div
          class="section"
        />
        <div
          class="moveButtons"
        >
          <span>
            ↑
          </span>
          <br />
          <br />
          <span>
            ↓
          </span>
        </div>
      </div>
    </div>
    <div
      class="addSectionContainer"
    >
      <p
        class="addSection"
      >
        ⊕ Add New Section
      </p>
    </div>
    <div
      class="footer"
    />
  </div>
</DocumentFragment>
`;
